pennylane/quantum_info/entropies.py

```diff
@@ -63,24 +63,57 @@ def _compute_vn_entropy(density_matrix, base=None):
     if interface == "jax":
         import jax
 
-        evs = jax.numpy.array([ev for ev in evs if ev >= 0])
+        evs = jax.numpy.array([ev for ev in evs if ev > 0])
         entropy = jax.numpy.sum(jax.scipy.special.entr(evs) / div_base)
 
     elif interface == "torch":
         import torch
 
-        evs = torch.tensor([ev for ev in evs if ev >= 0])
+        evs = torch.tensor([ev for ev in evs if ev > 0])
         entropy = torch.sum(torch.special.entr(evs) / div_base)
 
     elif interface == "tensorflow":
         import tensorflow as tf
 
         evs = tf.math.real(evs)
-        evs = tf.Variable([ev for ev in evs if ev >= 0])
+        evs = tf.Variable([ev for ev in evs if ev > 0])
         entropy = -tf.math.reduce_sum(evs * tf.math.log(evs) / div_base)
 
     else:
-        evs = qml.math.array([ev for ev in evs if ev >= 0])
+        evs = qml.math.array([ev for ev in evs if ev > 0])
```

Comment:
Could this or any of the other similar lines cause some issues with gradient flow?

Reply:
It will probably compute the gradient still but I think there _might_ be an issue when ev=0 but grad(ev)!=0. I commented something similar in [QIM-7](https://github.com/PennyLaneAI/pennylane/pull/2663#pullrequestreview-997061467).